gems/smithy/lib/smithy/plan.rb

```diff
@@ -31,5 +32,18 @@ def initialize(model, type, options = {})
 
     # @return [Array<Polish>] The polishes that apply to this plan.
     attr_reader :polishes
+
+    # @return [Hash<String, Hash>] The service shape for the shapes.
+    attr_reader :service
```

Comment:
Can this just be the service instead of a hash with a single entry?

Reply:
I think we also want the id for any lookup. So either the id or this hash.. I chose the hash because we would often have to look up based on that id anyway.

Reply:
The model parsing uses "shapes" which are the hashes. I think this makes sense to stay consistent for all shapes. Some times you need to re-reference via the ID. Service could work differently but then it's more confusing I think. So if we just expect a hash everywhere, it's easy to follow.

Reply:
I guess I was confused - in the current ServiceIndex the `service` returned is a Hash with a single key of the id and value of the service shape (a Hash with values of Hash) so code that used it needed to do something like `service.values.first`. Maybe RBS typing would help here, but I can't actually tell what this method returns (ie, is it a map to shape or is it the shape?)

Reply:
My understanding of "shape" was a hash with id to shape data, and multiple shapes are the same but flat (one hash with id to shape data). I tried to make all of the interfaces accept a hash even if it's one key (one shape). I did this because we often need the id. I'll think more on how to change this however I'm keeping it consistent for now.

Reply:
Ah I understand.  I find that a little hard to work - maybe RBS typing would help, but this is fine for now.

Reply:
Discussed offline - I think we NEED service id here for reconciling module names eventually. I can also make the documentation better.